test/e2e/network_segmentation.go

```diff
@@ -508,6 +509,97 @@ var _ = Describe("Network Segmentation", func() {
 		expectedMessage := fmt.Sprintf("primary network already exist in namespace %q: %q", f.Namespace.Name, primaryNadName)
 		Expect(actualConditions[0].Message).To(Equal(expectedMessage))
 	})
+
+	Context("pod2Egress on a user defined primary network", func() {
+		const (
+			externalContainerName = "ovn-k-egress-test-helper"
+		)
+		var externalIpv4, externalIpv6 string
+		BeforeEach(func() {
+			externalIpv4, externalIpv6 = createClusterExternalContainer(
+				externalContainerName,
+				"registry.k8s.io/e2e-test-images/agnhost:2.45",
+				runExternalContainerCmd(),
+				httpServerContainerCmd(port),
+			)
+
+			DeferCleanup(func() {
+				deleteClusterExternalContainer(externalContainerName)
+			})
+		})
+
+		DescribeTable(
+			"can be accessed to from the pods running in the Kubernetes cluster",
+			func(netConfigParams networkAttachmentConfigParams, clientPodConfig podConfiguration) {
+				netConfig := newNetworkAttachmentConfig(netConfigParams)
+
+				netConfig.namespace = f.Namespace.Name
+				clientPodConfig.namespace = f.Namespace.Name
+
+				By("creating the attachment configuration")
+				_, err := nadClient.NetworkAttachmentDefinitions(f.Namespace.Name).Create(
+					context.Background(),
+					generateNAD(netConfig),
+					metav1.CreateOptions{},
+				)
+				Expect(err).NotTo(HaveOccurred())
+
+				By("instantiating the client pod")
+				clientPod, err := cs.CoreV1().Pods(clientPodConfig.namespace).Create(
+					context.Background(),
+					generatePodSpec(clientPodConfig),
+					metav1.CreateOptions{},
+				)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(clientPod).NotTo(BeNil())
+
+				By("asserting the client pod reaches the `Ready` state")
+				var updatedPod *v1.Pod
+				Eventually(func() v1.PodPhase {
+					updatedPod, err = cs.CoreV1().Pods(f.Namespace.Name).Get(context.Background(), clientPod.GetName(), metav1.GetOptions{})
+					if err != nil {
+						return v1.PodFailed
+					}
+					return updatedPod.Status.Phase
+				}, 2*time.Minute, 6*time.Second).Should(Equal(v1.PodRunning))
+				framework.Logf("Client pod was created on node %s", updatedPod.Spec.NodeName)
+
+				By("asserting UDN pod is connected to UDN network")
+				podAnno, err := unmarshalPodAnnotation(updatedPod.Annotations, f.Namespace.Name+"/"+userDefinedNetworkName)
+				Expect(err).NotTo(HaveOccurred())
+				framework.Logf("Client pod's annotation for network %s is %v", userDefinedNetworkName, podAnno)
+				Expect(len(podAnno.Routes)).To(Equal(6)) // 3 v4 routes + 3 v6 routes for UDN
```

Comment:
I don't remember writing this expectation.

Isn't this L3 specific ? For sure I would expect to have less routes on L2. 

I would also expect to see even less routes for IPv4 only clusters.

And please use the `HaveLen` matcher instead of expecting the length of an array to be X - the error 
```
  [FAILED] Expected
      <int>: 3
  to equal
      <int>: 6
```
is not informative.

Reply:
Well, let's postpone this; I can address it in my layer2 topology PR. 

Are you OK with that ?

Reply:
I think this route checking was the result of a request from @trozet: https://github.com/ovn-org/ovn-kubernetes/pull/4557#discussion_r1711823486

Sounds good to mee to address it in the L2 PR.

Reply:
that is correct, it came from the fact that we wanted to check routes were there, surely we can make it better in future by grepping for exact routes..

Reply:
and I knew we were only running dualstack upstream which is why I did that, but I should add a check for if IPV6 {} and then do the needed.
but I want to get the ds merge in first so that we can rebase this on master

Reply:
OK. I agree we should assert the routes are the expected ones on a follow-up.